File: sort-an-array/sort-an-array.cpp
class Solution {

    
    void merge(vector<int>& nums, int l, int m, int r){
        vector<int> tmp(r - l + 1);
        int i = l; // index for left subarray
        int j = m + 1; // index for right subarray
        int k = 0; // index for temporary array
        while(i <= m && j <= r){
            if(nums[i] <= nums[j]) tmp[k++] = nums[i++]; 
            else tmp[k++] = nums[j++];
        }
        while(i <= m) tmp[k++] = nums[i++];
        while(j <= r) tmp[k++] = nums[j++]; 
        for(i = 0; i < k; i++) nums[l + i] = tmp[i];
    }
	
    void merge_sort(vector<int>&v,int l,int r){
        if(l<r){
        int mid=l+(r-l)/2;
        merge_sort(v,l,mid);
        merge_sort(v,mid+1,r);
        merge(v,l,mid,r);
        }
    }
    
public:
    vector<int> sortArray(vector<int>& nums) {
        merge_sort(nums,0,nums.size()-1);
        return nums;
    }
};

File: 948-sort-an-array/sort-an-array.py
class Solution:
    # selection sort
    # def sortArray(self, nums: List[int]) -> List[int]:
    #     # selection sort
    #     for i in range(len(nums)):
    #         min_ind = i
    #         for j in range(i,len(nums)):
    #             if(nums[min_ind]>nums[j]):
    #                 min_ind = j
    #         nums[i], nums[min_ind] = nums[min_ind], nums[i]
    #     return nums


    # def sortArray(self, nums: List[int]) -> List[int]:
    #     n = len(nums)
    #     for i in range(n):
    #         for j in range(n-i-1):
    #             if nums[j]>nums[j+1]:
    #                 nums[j], nums[j+1] = nums[j+1], nums[j]
    #     return nums

    def mergeSort(self,arr:list):
        if len(arr)>1:
            mid = len(arr)//2
            L = arr[:mid]
            R = arr[mid:]
            self.mergeSort(L)
            self.mergeSort(R)
            i=j=k=0
            while i<len(L) and j<len(R):
                if L[i]<R[j]:
                    arr[k] = L[i]
                    i+=1
                else:
                    arr[k] = R[j]
                    j+=1
                k+=1
            while i<len(L):
                arr[k] = L[i]
                i+=1
                k+=1

            while j<len(R):
                arr[k] = R[j]
                j+=1
                k+=1
             
    def sortArray(self, nums: List[int]) -> List[int]:
        self.mergeSort(nums)
        return nums